From 6aa88a15295b68ef3c8f0a489468a9b30e5637f4 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:53 -0400
Subject: [powerpc] powernv: Change BUG_ON to WARN_ON in elog code

Message-id: <1410545655-205645-605-git-send-email-dzickus@redhat.com>
Patchwork-id: 94666
O-Subject: [RHEL7 PATCH 604/626] powerpc/powernv: Change BUG_ON to WARN_ON in elog code
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit fa952c54ba13deeb91cd4c7af255cdb5f1273535
Author: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
Date:   Wed Jul 23 14:52:39 2014 +0530

    powerpc/powernv: Change BUG_ON to WARN_ON in elog code

    We can continue to read the error log (up to MAX size) even if
    we get the elog size more than MAX size. Hence change BUG_ON to
    WARN_ON.

    Also updated error message.

    Reported-by: Gopesh Kumar Chaudhary <gopchaud@in.ibm.com>
    Signed-off-by: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
    Signed-off-by: Ananth N Mavinakayanahalli <ananth@in.ibm.com>
    Acked-by: Deepthi Dharwar <deepthi@linux.vnet.ibm.com>
    Acked-by: Stewart Smith <stewart@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/opal-elog.c b/arch/powerpc/platforms/powernv/opal-elog.c
index b823afb..30b0e65 100644
--- a/arch/powerpc/platforms/powernv/opal-elog.c
+++ b/arch/powerpc/platforms/powernv/opal-elog.c
@@ -254,7 +254,7 @@ static void elog_work_fn(struct work_struct *work)
 
  rc = opal_get_elog_size(&id, &size, &type);
  if (rc != OPAL_SUCCESS) {
-  pr_err("ELOG: Opal log read failed\n");
+  pr_err("ELOG: OPAL log info read failed\n");
   return;
  }
 
@@ -262,7 +262,7 @@ static void elog_work_fn(struct work_struct *work)
  log_id = be64_to_cpu(id);
  elog_type = be64_to_cpu(type);
 
- BUG_ON(elog_size > OPAL_MAX_ERRLOG_SIZE);
+ WARN_ON(elog_size > OPAL_MAX_ERRLOG_SIZE);
 
  if (elog_size >= OPAL_MAX_ERRLOG_SIZE)
   elog_size  =  OPAL_MAX_ERRLOG_SIZE;
-- 
1.7.1